### ISSUE TITLE
Autest for HTTP/3 isn't ran on the CI job

### ISSUE THREAD

**Issue by maskit** [apache/trafficserver#9067]
We have at least one autest that checks HTTP/3, and I confirmed it was ran when we close #3551.

There should be four dots on active_timeout test if the test for HTTP/3 is ran, but the latest log only shows three dots.
```
Running Test active_timeout:... Passed
```

It's either ATS or curl (or the both) is not built with an appropriate library.
https://github.com/apache/trafficserver/blob/1f7799c2b66a6a57709fa9f1176774da57c778fc/tests/gold_tests/timeout/active_timeout.test.py#L67-L70

**Comment by bneradt** [comment]
That's interesting. I wonder how this broke in the last few months.

Here is our pipeline script:
https://github.com/apache/trafficserver-ci/blob/main/jenkins/github/autest.pipeline

We try to do the right thing: building against `/opt/openssl-quic` and using `/opt/bin` to pick up our quic version of openssl.

Thank you for pointing this out.

**Comment by bneradt** [comment]
I created a docker container with our CI. It looks like ATS has the QUIC feature built in:

```
[root@97e217d348ae tests]# /tmp/ats/bin/traffic_layout --features --json | grep QUIC
    "TS_USE_QUIC": 1,
```

But curl somehow no longer advertises `http3` support:

```
[root@97e217d348ae tests]# which curl
/opt/bin/curl
[root@97e217d348ae tests]# curl --version
curl 7.79.0-DEV (x86_64-pc-linux-gnu) libcurl/7.79.0-DEV OpenSSL/1.1.1m zlib/1.2.11 brotli/1.0.6 zstd/1.4.4 nghttp2/1.45.0-DEV
Release-Date: [unreleased]
Protocols: dict file ftp ftps gopher gophers http https imap imaps mqtt pop3 pop3s rtsp smb smbs smtp smtps telnet tftp 
Features: alt-svc AsynchDNS brotli HSTS HTTP2 HTTPS-proxy IPv6 Largefile libz NTLM NTLM_WB SSL TLS-SRP UnixSockets zstd
[root@97e217d348ae tests]# curl --version | grep http3
[root@97e217d348ae tests]# 
```

So the problem is with the CI's version of curl. I'll take this opportunity to update our OpenSSL version on CI and see whether I can fix our built version of curl to have `http3` support again.

**Comment by bneradt** [comment]
I've created a version of `ci.trafficserver.apache.org/ats/rockylinux:8` that has openssl-quic updated to 1.1.1q and has a version of curl correctly built with HTTP3 support.

Unfortunately, I cannot instantiate the image into our CI system because the `active_timeout` QUIC test fails (the one corresponding to the fourth dot in the autest output). It looks like curl is failing to connect to ATS: 

```
[root@9086c9d23446 tests]# cat /tmp/sandbox/active_timeout/_output/3-tr-Default/stream.all.txt 
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (56) Failed to connect to 127.0.0.1 port 61002 after 3 ms: Failure when receiving data from the peer
```

**Comment by bneradt** [comment]
Adding the `quic` debug tag, I see ATS emits this on the incoming connection:

```
[Aug 29 16:15:15.357] [ET_UDP 0] DEBUG: <QUICPacketHandler.cc:260 (_recv_packet)> (quic_sec) [82a4202c-2ee6de91] Unsupported version: 0x1
```

This sounds related to the `QUIC status` thread in the mailing list.

**Comment by maskit** [comment]
Ah, right. The latest version of curl probably only uses QUICv1 (the final version but not a draft version) and there's no way to specify a QUIC version to use. Even after we merge 10-Dev to master, 9.2.x and 9.1.x will still have the version mismatch issue unless we add the support to it.

What do we want to do? Should I back port the support to 9.x?

**Comment by bneradt** [comment]
OK, I've updated the CI images with the latest version of openssl-quic 1.1.1q with the updated curl as well. Because 9.x doesn't support QUICv1 (as @maskit points out above) I disabled the tests on 9.x builds because that doesn't support the latest tooling.

Once 10-Dev is fixed for this new openssl version, we can make sure the following:

1. 10-Dev has the latest master merged into it in such a way that the active_timeout still runs the QUIC TestRun.
2. Make sure the test passes.

Once those two things are done, then we can close this issue.
